src/p4est_io.h

```diff
@@ -68,6 +108,398 @@ p4est_t            *p4est_inflate (sc_MPI_Comm mpicomm,
+ * \param [out] errcode       An errcode that can be interpreted by \ref
+ *                            p4est_file_error_string and
+ *                            \ref p4est_file_error_class.
+ * \return                    Return a pointer to input context or NULL in case
+ *                            of errors that does not abort the program or if
+ *                            the function was called with quadrant_data == NULL.
+ *                            In case of error the file is tried to close
+ *                            and fc is freed.
+ */
+p4est_file_context_t *p4est_file_read_field (p4est_file_context_t * fc,
+                                             sc_array_t * quadrant_data,
+                                             char *user_string, int *errcode);
+
+/** A data type that encodes the metadata of one data block in a p4est data file.
+ */
+typedef struct p4est_file_block_metadata
+{
+  char                block_type; /**< 'H' (header) or 'F' (data file) */
+  size_t              data_size;  /**< data size in bytes per array element ('F')
+                                       or of the header block ('H') */
+  char                user_string[48]; /**< user string of the data block */
+}
+p4est_file_block_metadata_t;
+
+/** Read metadata information of a file written by a matching forest.
+ * Matching refers to the global count of quadrants; partition is irrelevant.
+ *
+ * This function parses the given file on rank 0 and broadcasts the information
+ * on the number of data fields contained to all other ranks.  Collective call.
+ *
+ * This function catches all I/O and file format errors and returns a valid MPI
+ * error class related to file handling.  Errors are collectively synchronized.
+ *
+ * If the number of bytes that the user intend to read is larger than the number
+ * bytes left in the file, the function prints out an information about this
+ * situation using \ref P4EST_LERROR. In this case the function reads the bytes
+ * that are possible to read but returns NULL to indicate an error.
+ * If the file or block header information is not matching the passed parameters
+ * the function sets \ref P4EST_ERR_IO for errcode.
+ *
+ * \param [in]  p4est               A p4est that is only required for the
+ *                                  MPI communicator, and to verify the
+ *                                  global quadrant count found in the file.
+ * \param [in]  filename            Path to parallel file.
+ * \param [in,out] user_string      At least 16 bytes. This array will
+ *                                  be filled with the user string of the
+ *                                  file after a successful call of this
+ *                                  function.
+ * \param [in,out] blocks           After a successful function call this
+ *                                  variable holds an array with a length
+ *                                  corresponding to the number of arrays in the
```

Comment:
The number of arrays in the file, or rather the number of elements (maybe not) or data pieces (maybe not) or data blocks (hmm)? Wondering what's the best term.

Reply:
Let's call each section of the file (i. e. the file metadata, each header, and each field) a "section", officially.

Reply:
I renamed the data blocks to data sections.